backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/OrganizationExtractor.java

```diff
@@ -25,35 +33,88 @@ public OrganizationExtractor(AuthorizationProperties properties) {
     this.properties = properties;
   }
 
-  @Override
-  public List<OrganizationRoleClaims> convert(Collection<? extends GrantedAuthority> source) {
+  /**
+   * There are two kinds of claims we care about - role claims and facility claims. Role claims take
+   * the format SR-<ENV>-TENANT:<ORG_EXTERNAL_ID>:<ROLE> Facility claims take the format
+   * SR-<ENV>-TENANT:<ORG_EXTERNAL_ID>:FACILITY_ACCESS:<FACILITY_UUID>
+   */
+  public List<OrganizationRoleClaims> convertClaims(Collection<String> claims) {
+    // Map of orgs to facilities the user can access therein
+    Map<String, Set<UUID>> facilitiesFound = new HashMap<>();
+    // Map of orgs to roles the user has therein
     Map<String, EnumSet<OrganizationRole>> rolesFound = new HashMap<>();
-    for (GrantedAuthority granted : source) {
-      String claimed = granted.getAuthority();
-      if (!claimed.startsWith(properties.getRolePrefix())) {
-        continue;
-      }
-      int roleOffset = claimed.lastIndexOf(":");
-      String claimedOrg = claimed.substring(properties.getRolePrefix().length(), roleOffset);
-      String claimedRole =
-          claimed.substring(roleOffset + 1); // the separator is part of neither string
+    for (String claimed : claims) {
       try {
-        OrganizationRole claimedRoleValidated = OrganizationRole.valueOf(claimedRole);
-        EnumSet<OrganizationRole> existingRoles = rolesFound.get(claimedOrg);
-        if (existingRoles == null) {
-          rolesFound.put(claimedOrg, EnumSet.of(claimedRoleValidated));
+        if (!claimed.startsWith(properties.getRolePrefix())) {
+          continue;
+        }
+        if (claimed.contains(FACILITY_ACCESS_MARKER)) {
+          int facilityOffset = claimed.lastIndexOf(CLAIM_SEPARATOR);
+          String claimedFacility = claimed.substring(facilityOffset + CLAIM_SEPARATOR.length());
+          int facilityMarkerOffset = claimed.lastIndexOf(CLAIM_SEPARATOR + FACILITY_ACCESS_MARKER);
```

Comment:
```suggestion
          int facilityMarkerOffset = facilityOffset - (FACILITY_ACCESS_MARKER.length() + CLAIM_SEPARATOR.length();
```

This relies on the assumption that the string immediately before the last ":" is ":FACILITY_ACCESS", but then if it's not you've got something reeeally weird going on, and it's going to break anyway—I'm not currently able to come up with scenarios where one of them is less catastrophically broken than the other.

In general I think this whole thing would benefit from our just imposing the rule that you can't have colons in identifiers, and using String.split(":"), but OTOH defensive coding is good. Which is to say, if you're strongly of the opinion that the original is less breakable than the suggested change, ignore this with my good will. 🙂